src/compiler/transformers/declarations.ts

```diff
@@ -511,64 +470,47 @@ export function transformDeclarations(context: TransformationContext) {
         suppressNewDiagnosticContexts = false;
         lateMarkedStatements = undefined;
         lateStatementReplacementMap = new Map();
-        necessaryTypeReferences = undefined;
-        refs = collectReferences(currentSourceFile, new Map());
-        libs = collectLibs(currentSourceFile, new Map());
-        const references: FileReference[] = [];
+        rawReferencedFiles = new Map();
+        rawTypeReferenceDirectives = new Map();
+        rawLibReferenceDirectives = new Map();
+        collectReferencedFiles(currentSourceFile, rawReferencedFiles);
+        collectTypeReferenceDirectives(currentSourceFile, rawTypeReferenceDirectives);
+        collectLibReferenceDirectives(currentSourceFile, rawLibReferenceDirectives);
+        // TODO(jakebailey): simplify
+        const referencedFiles: FileReference[] = [];
         const outputFilePath = getDirectoryPath(normalizeSlashes(getOutputPathsFor(node, host, /*forceDtsPaths*/ true).declarationFilePath!));
-        const referenceVisitor = mapReferencesIntoArray(references, outputFilePath);
+        const referenceVisitor = mapReferencesIntoArray(referencedFiles, outputFilePath);
         let combinedStatements: NodeArray<Statement>;
         if (isSourceFileJS(currentSourceFile)) {
             combinedStatements = factory.createNodeArray(transformDeclarationsForJS(node));
-            refs.forEach(referenceVisitor);
-            emittedImports = filter(combinedStatements, isAnyImportSyntax);
+            rawReferencedFiles.forEach(referenceVisitor);
         }
         else {
             const statements = visitNodes(node.statements, visitDeclarationStatements, isStatement);
             combinedStatements = setTextRange(factory.createNodeArray(transformAndReplaceLatePaintedStatements(statements)), node.statements);
-            refs.forEach(referenceVisitor);
-            emittedImports = filter(combinedStatements, isAnyImportSyntax);
+            rawReferencedFiles.forEach(referenceVisitor);
             if (isExternalModule(node) && (!resultHasExternalModuleIndicator || (needsScopeFixMarker && !resultHasScopeMarker))) {
                 combinedStatements = setTextRange(factory.createNodeArray([...combinedStatements, createEmptyExports(factory)]), combinedStatements);
             }
         }
-        const updated = factory.updateSourceFile(node, combinedStatements, /*isDeclarationFile*/ true, references, getFileReferencesForUsedTypeReferences(), node.hasNoDefaultLib, getLibReferences());
+        const updated = factory.updateSourceFile(node, combinedStatements, /*isDeclarationFile*/ true, referencedFiles, getTypeReferences(), node.hasNoDefaultLib, getLibReferences());
         updated.exportedModulesFromDeclarationEmit = exportedModulesFromDeclarationEmit;

```

Comment:
I haven't removed this prop yet, but it seems like it probably shouldn't be needed anymore? I think this is used for build mode / incremental and now that the determination of whether or not references are included is entirely file local, I don't think it's possible for us to need to keep track of anything else anymore?